newIDE/app/src/RedemptionCode/RedemptionCodesDialog.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+import AuthenticatedUserContext from '../Profile/AuthenticatedUserContext';
+import { LineStackLayout } from '../UI/Layout';
+import RedemptionCodeIcon from '../UI/CustomSvgIcons/RedemptionCode';
+import Text from '../UI/Text';
+import Dialog from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import {
+  getRedemptionCodes,
+  type RedemptionCode,
+} from '../Utils/GDevelopServices/Usage';
+import PlaceholderLoader from '../UI/PlaceholderLoader';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn,
+} from '../UI/Table';
+import {
+  getPlanIcon,
+  getPlanInferredNameFromId,
+} from '../Profile/Subscription/PlanCard';
+import AlertMessage from '../UI/AlertMessage';
+
+type Props = {|
+  onClose: () => void,
+|};
+
+const RedemptionCodesDialog = ({ onClose }: Props) => {
+  const { profile, getAuthorizationHeader } = React.useContext(
+    AuthenticatedUserContext
+  );
+  const [
+    redemptionCodes,
+    setRedemptionCodes,
+  ] = React.useState<?Array<RedemptionCode>>(null);
+
+  React.useEffect(
+    () => {
+      if (profile) {
```

Comment:
Ubernitpicking, but prefer early return `if (!profile) return;` to avoid useless indentations